app/models/user.rb

```diff
@@ -57,6 +60,33 @@ def self.to_csv
     end
   end
 
+  def self.from_omniauth(access_token)
+    data = access_token[:info]
+    user = User.find_by(email: data[:email])
+    unless user
+      password = Devise.friendly_token[0,20]
+      user = User.create(first_name: data[:first_name],
+        last_name: data[:last_name],
+        email: data[:email],
+        role: ADMIN,
+        google_token: access_token[:credentials][:token],
+        google_refresh_token: access_token[:credentials][:refresh_token]
+      )
+    end
+    user
+  end
+
+  def self.new_with_session(params, session)
+    super.tap do |user|
+      if data = session["devise.google_data"]
+        user.email = data["info"]["email"] if user.email.blank?
```

Comment:
try to avoid conditional assignments
u could use `Hash#dig` instead, like
```
if session["devise.google_data"].present?
  user.email = session.dig('devise.google_data', 'info', 'email') if user.email.blank?

  # ...
end
```

Reply:
так використай метод dig, заодно вивчиш його